File: OOP_Variable/4.py
class C(object) :
    def __init__(self, v) :
        self.__value = v #__value로하면 인스턴스 밖에서는 접근 불가
    def show(self) :
        print(self.__value)

c1 = C(10)

#print(c1.__value) #직접 접근은 불가능
c1.show() # 메소드를 통한 접근은 가능

File: OOP_Variable/4.rb
class C
    #attr_reader :value
    #attr_writer :value
    attr_accessor :value #reader, write 모두 허용
    def initialize(v)
        @value = v
    end
    def show()
        p(@value)
    end
end

c1 = C.new(10) #루비에서는 외부에서 인스턴스 변수 접근 불가능
p(c1.value)
c1.value = 20
p(c1.value)
#c1.setValue(20)
#c1.show()